go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -0,0 +1,719 @@
+package ovn
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	networkattachmentdefinitionapi "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/metrics"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/ipallocator"
+	ovntypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	"github.com/pkg/errors"
+	kapi "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	utilnet "k8s.io/utils/net"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+	"github.com/ovn-org/libovsdb/ovsdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+)
+
+func (bnc *BaseNetworkController) updateExpectedLogicalPortsAndReservePodIPs(pod *kapi.Pod,
+	annotations *util.PodAnnotation, expectedLogicalPorts map[string]bool) error {
```

Comment:
I guess I did not notice the expectedLogicalPorts map before. It looks a bit strange. Can't this method just return a boolean instead of receiving and updating a map? And then the method could be `allocatePodIPsIfRequired`.

Or even better, we could have a `podRequiresOvnNetwork` that returns a boolean as is self-explanatory, and if it does then we can call this method as `allocatePodIPs`.

Reply:
sure